### PR TITLE
Add implementation of standard deviation on data

### PR DESCRIPTION
**Pull request by yv4920**
Extend the graphs to show standard deviation

### CODE REVIEW

File: inflammation/models.py [hunk 1]
```diff
@@ -54,3 +54,13 @@ def daily_min(data):
...
 
+
+def s_dev(data):
+    """Computes and returns standard deviation for data."""
+    mmm = np.mean(data, axis=0)
+    devs = []
+    for entry in data:
+        devs.append((entry - mmm) * (entry - mmm))
+
+    s_dev2 = sum(devs) / len(data)
```
Comment (lornasibson) [inline inflammation/models.py]: Could just use built in numpy function to calculate standard deviation

File: inflammation/models.py [hunk 2]
```diff
@@ -54,3 +54,13 @@ def daily_min(data):
...
+    return {'standard deviation': s_dev2}
```
Comment (lornasibson) [inline inflammation/models.py]: Could just return the standard deviation instead of a dictionary - will make testing and debugging much easier

File: inflammation/models.py [hunk 3]
```diff
@@ -54,3 +54,13 @@ def daily_min(data):
     """Calculate the daily min of a 2d inflammation data array."""
     return np.min(data, axis=0)
 
+
+def s_dev(data):
+    """Computes and returns standard deviation for data."""
```
Comment (lornasibson) [inline inflammation/models.py]: Documentation is incomplete, should include input parameters and returns to the function. Also, currently the function is returning a dictionary, not the standard deviation

File: inflammation-analysis.py [hunk 4]
```diff
@@ -27,7 +27,8 @@ def main(args):
     for filename in InFiles:
         inflammation_data = models.load_csv(filename)
 
-        view_data = {'average': models.daily_mean(inflammation_data), 'max': models.daily_max(inflammation_data), 'min': models.daily_min(inflammation_data)}
+        view_data = {'average': models.daily_mean(inflammation_data), 'max': models.daily_max(inflammation_data), 'min': models.daily_min(inflammation_data), **(models.s_dev(inflammation_data))}
```
Comment (lornasibson) [inline inflammation-analysis.py]: Line is too long, split the line into multiple lines to make the code more readable

File: tests/test_models.py [hunk 5]
```diff
@@ -37,3 +37,14 @@ def test_load_from_json(tmpdir):
...
     npt.assert_array_equal(result, [[1, 2, 3], [4, 5, 6]])
+
+
+@pytest.mark.parametrize('data, expected_standard_deviation', [
+    ([0, 0, 0], 0.0),
+    ([1.0, 1.0, 1.0], 0),
+    ([0.0, 2.0], 1.0)
+])
+def test_daily_standard_deviation(data, expected_standard_deviation):
+    from inflammation.models import s_dev
```
Comment (lornasibson) [inline tests/test_models.py]: Move import statement to top of file

File: inflammation/models.py [hunk 6]
```diff
@@ -54,3 +54,13 @@ def daily_min(data):
     """Calculate the daily min of a 2d inflammation data array."""
     return np.min(data, axis=0)
 
+
+def s_dev(data):
```
Comment (lornasibson) [inline inflammation/models.py]:
```suggestion
def standard_deviation(data):
```

File: inflammation/models.py [hunk 7]
```diff
@@ -54,3 +54,13 @@ def daily_min(data):
...
+    """Computes and returns standard deviation for data."""
+    mmm = np.mean(data, axis=0)
```
Comment (lornasibson) [inline inflammation/models.py]:
```suggestion
    means = np.mean(data, axis=0)
```

File: inflammation/models.py (hunk 1 above)
Comment (lornasibson) [inline inflammation/models.py]:
```suggestion
    std_dev = sum(devs) / len(data)
```

**Review by lornasibson — CHANGES_REQUESTED** [review]
Expected tests:
- Edge case for if `data` is empty
- Edge case for if `data` only contains 1 value
- Testing adding std_dev dict to view_data
- Test adding to view_data with an empty std_dev dict
- Test that graph includes standard deviation graph
- Standard deviation function should throw error if given empty data variable
- Standard deviation function should work with negative inputs
